.github/actions/upload-docs/action.yaml

```diff
@@ -10,20 +10,29 @@ inputs:
     description: "Name of the deployment environment"
     required: false
     default: "github-pages"
+  script:
+    description: 'The script to run for generating documentation'
+    required: false
+    default: 'scripts/generateOverviewDocs.js'
     
 runs:
   using: "composite"
   steps:
-    - name: Install Solidity docgen
-      run: npm install -g solidity-docgen
-      shell: bash
-
     - name: Create docs directory
-      run: mkdir -p docs
+      run: |
+        if [ -d "docs" ]; then
+          rm -rf docs
+          echo "removed pre-existing 'docs' dir"
+        fi
+        mkdir -p docs
       shell: bash
 
     - name: Build Solidity documentation
-      run: npx hardhat docgen
+      run: |
+        if ! npx hardhat docgen; then
+          echo "Hardhat failed to generate documentation"
+          exit 1
+        fi
       shell: bash
 
     - name: Check for generateOverviewDocs script
```

Comment:
Avoid this specialized logic. Just provide the script in this repo as input to this action as well